### PR TITLE
Get rid of @ts-ignore

### PR DESCRIPTION
**Pull request by gianantoniopini**
This pull request is to get rid of some of the `// @ts-ignore` in the code.
It is somehow related to issue #76 , however there are still some `// @ts-ignore` left.

### CODE REVIEW

**Comment by gianantoniopini** [comment]
Hi @bishoymelek @HanaAyman @roosnic1, could you please have a look at this pull request ?

File: packages/api/src/server.ts [hunk 1]
```diff
@@ -39,12 +39,12 @@ export class WepublishServer {
 
     methodsToProxy.forEach(mtp => {
       if (mtp.key in dbAdapter) {
+        const dbAdapterKeyTyped = mtp.key as keyof typeof dbAdapter
```
Comment (bishoymelek) [inline packages/api/src/server.ts:42]: I think this file changes needs more enhancement, as this line isn't satisfying for line 47 for example.

**Review by bishoymelek — CHANGES_REQUESTED** [review]
Thank you @gianantoniopini for the contribution. I just left small comments and will debug it later this week, if still pending. thanks.

**Comment by gianantoniopini** [comment]
Thank you @bishoymelek for the review. I have started to address the requested changes.

File: packages/api-db-mongodb/src/db/invoice.ts [hunk 2]
```diff
@@ -233,4 +232,6 @@ function invoiceDateForSort(invoice: DBInvoice, sort: InvoiceSort): Date {
     case InvoiceSort.PaidAt:
       if (invoice.paidAt) return invoice.paidAt
   }
+
+  return undefined
```
[the remark this answers is not in the document]
Comment (gianantoniopini) [reply packages/api-db-mongodb/src/db/invoice.ts]: @bishoymelek Yes, it is a bit more clear with default case. I've just changed it.

File: packages/api/src/server.ts (hunk 1 above)
Comment (gianantoniopini) [reply packages/api/src/server.ts:42]: Yes the `// @ts-ignore` for line 47 is still there as I could not find a simple way to get rid of that. I will investigate further.

Comment (gianantoniopini) [reply packages/api/src/server.ts:42]: Not sure how to get rid of the `// @ts-ignore` for line 47, as this one is a bit beyond my current knowledge of TypeScript. I believe it is not straightforward as `methodName` is only known at runtime and it is used as a deep key of the `dbAdapter` object. @bishoymelek Perhaps you have some ideas?

File: packages/api-db-mongodb/src/db/invoice.ts [hunk 3]
```diff
@@ -232,5 +231,9 @@ function invoiceDateForSort(invoice: DBInvoice, sort: InvoiceSort): Date {
 
     case InvoiceSort.PaidAt:
       if (invoice.paidAt) return invoice.paidAt
```
Comment (roosnic1) [inline packages/api-db-mongodb/src/db/invoice.ts:233]: I'd recommend something like `return invoice.paidAt ? invoice.paidAt : invoice.createdAt` here. With this you don't need the `... | undefined` on line 224.

File: packages/api/src/server.ts (hunk 1 above)
Comment (roosnic1) [reply packages/api/src/server.ts:42]: @gianantoniopini: this whole implementation should get refactored anyway so I am already happy if we can get rid of the `// @ts-ignore` you did 😃 .

**Review by roosnic1 — CHANGES_REQUESTED** [review]
@gianantoniopini thanks so much 🎉 . Looks very good just one little change and then we can merge it.

File: packages/api-db-mongodb/src/db/invoice.ts (hunk 3 above)
Comment (gianantoniopini) [reply packages/api-db-mongodb/src/db/invoice.ts]: Thanks. Just changed it as per your recommendation. I've also changed the `switch` `default` case to `return invoice.createdAt`. Happy to apply more changes if needed.

**Comment by gianantoniopini** [comment]
@roosnic1 thanks a lot for reviewing this 😃 . I've just applied the change as per your recommendation. Let me know in case more tweaks are required.

**Review by roosnic1 — APPROVED** [review]
🎉 awesome thank you very much

**Review by bishoymelek — APPROVED** [review]
Thanks a lot @gianantoniopini 💯 🎉

**Review by penk85 — APPROVED** [review]
:+1:

**Review by camilleory — APPROVED** [review]

**Review by lakeside763 — APPROVED** [review]

**Pull request merged**